dpnp/tests/test_statistics.py

```diff
@@ -127,6 +127,186 @@ def test_avg_error(self):
+            pytest.skip("avoid overflow.")
+        an = generate_random_numpy_array(
+            a_size, dtype, low=-3, high=3, probability=0.9
+        )
+        vn = generate_random_numpy_array(
+            v_size, dtype, low=-3, high=3, probability=0.9
+        )
+
+        ad = dpnp.array(an)
+        vd = dpnp.array(vn)
+
+        dpnp_kwargs, numpy_kwargs = self._get_kwargs(mode, method)
+
+        result = dpnp.convolve(ad, vd, **dpnp_kwargs)
+        expected = numpy.convolve(an, vn, **numpy_kwargs)
+
+        if method != "fft" and (
+            dpnp.issubdtype(dtype, dpnp.integer) or dtype == dpnp.bool
+        ):
+            # For 'direct' and 'auto' methods, we expect exact results for integer types
+            assert_array_equal(result, expected)
+        else:
+            if method == "direct":
+                # For 'direct' method we can use standard validation
+                # acceptable error depends on the kernel size
+                # while error grows linearly with the kernel size,
+                # this empirically found formula provides a good balance
+                # the resulting factor is 40 for kernel size = 1,
+                # 400 for kernel size = 100 and 4000 for kernel size = 10000
+                factor = int(40 * (min(a_size, v_size) ** 0.5))
+                assert_dtype_allclose(result, expected, factor=factor)
+            else:
+                rdtype = result.dtype
+                if dpnp.issubdtype(rdtype, dpnp.integer):
+                    # 'fft' do its calculations in float
+                    # and 'auto' could use fft
+                    # also assert_dtype_allclose for integer types is
+                    # always check for exact match
+                    rdtype = dpnp.default_float_type(ad.device)
+
+                result = result.astype(rdtype)
+
+                rtol = 1e-3
+                atol = 1e-10
+
+                if rdtype == dpnp.float64 or rdtype == dpnp.complex128:
+                    rtol = 1e-6
+                    atol = 1e-12
+                elif rdtype == dpnp.bool:
+                    result = result.astype(dpnp.int32)
+                    rdtype = result.dtype
+
+                expected = expected.astype(rdtype)
+
+                diff = numpy.abs(result.asnumpy() - expected)
+                invalid = diff > atol + rtol * numpy.abs(expected)
+
+                # When using the 'fft' method, we might encounter outliers.
+                # This usually happens when the resulting array contains values close to zero.
+                # For these outliers, the relative error can be significant.
+                # We can tolerate a few such outliers.
+                max_outliers = 8 if expected.size > 1 else 0
+                if invalid.sum() > max_outliers:
+                    # we already failed check,
+                    # call assert_dtype_allclose just to report error nicely
+                    assert_dtype_allclose(result, expected, factor=1000)
```

Comment:
it is supposed to raise an error regardless of the value `factor`?
```suggestion
                    assert_dtype_allclose(result, expected)
```

Reply:
No. In order to correctly report an error `factor` provided to `assert_dtype_allclose` should generate similar tolerance.